stubs/tensorflow/tensorflow/linalg.pyi

```diff
@@ -0,0 +1,50 @@
+from _typeshed import Incomplete
+from builtins import bool as _bool
+from typing import Literal, overload
+
+from tensorflow import RaggedTensor, Tensor, norm as norm
+from tensorflow._aliases import DTypeLike, ScalarTensorCompatible, TensorCompatible
+from tensorflow.math import l2_normalize as l2_normalize
+
+@overload
+def matmul(
+    a: TensorCompatible,
+    b: TensorCompatible,
+    transpose_a: _bool = False,
+    transpose_b: _bool = False,
+    adjoint_a: _bool = False,
+    adjoint_b: _bool = False,
+    a_is_sparse: _bool = False,
+    b_is_sparse: _bool = False,
+    output_type: DTypeLike | None = None,
+    name: str | None = None,
+) -> Tensor: ...
+@overload
+def matmul(
+    a: RaggedTensor,
+    b: RaggedTensor,
+    transpose_a: _bool = False,
+    transpose_b: _bool = False,
+    adjoint_a: _bool = False,
+    adjoint_b: _bool = False,
+    a_is_sparse: _bool = False,
+    b_is_sparse: _bool = False,
+    output_type: DTypeLike | None = None,
+    name: str | None = None,
+) -> RaggedTensor: ...
+def set_diag(
+    input: TensorCompatible,
+    diagonal: TensorCompatible,
+    name: str | None = "set_diag",
+    k: int = 0,
+    align: Literal["RIGHT_LEFT", "RIGHT_RIGHT", "LEFT_LEFT", "LEFT_RIGHT"] = "RIGHT_LEFT",
+) -> Tensor: ...
+def eye(
+    num_rows: ScalarTensorCompatible,
+    num_columns: ScalarTensorCompatible | None = None,
+    batch_shape: int | list[int] | tuple[int, ...] | None = None,
```

Comment:
Minor nit: I'd lean to relax to TensorCompatible. batch_shape is also compatible with tensor, `A list or tuple of Python integers or a 1-D int32 Tensor`. Looking over implementation I think `Iterable[int] | TensorCompatible` is about as broad as it can be. It's very rare for list[int] to be compatible but not int tensor.

Reply:
`TensorCompatible` includes invalid types, like `str`, `float` and array of strings/float. I feel like `TensorCompatible` is so broad that it is rarely the best match.

`Iterable[int] | IntArray | tf.Tensor` seems better to me, so I went with that in 6e2d07a41.